server/src/middlewares/user.js

```diff
@@ -0,0 +1,26 @@
+import { signUp as validateSignup, signIn as validateSignIn } from '../utils/validations';
+
+const signUpFieldsValid = async (req, res, next) => {
```

Comment:
Common wrapper which will take, user and function and return error and isValid. The benefit of the abstractions is that you can easily test the abstractions without passing the req, res, next.

Reply:
For the middlewares, I guess I need to use the req, res and next parameters?  Because without that I think it won't make sense?